### ISSUE TITLE
Port Drawing from Cairo to GtkSnapshot API

### ISSUE THREAD

**Issue by A6GibKm** [vixalien/decibels#61]
While the current code works, Cairo is drawn on the CPU rather than the GPU which might be less performant. On a more important note, Cairo uses physical pixels rather than logical pixels for its sizing. This can make drawing blurry while using GDK_SCALE=2 (e.g. for HIDPI screens).

**Comment by vixalien** [comment]
You're right. This is something I'm looking forward to, using Gtk4 paths et, all. Will keep you posted!

**Comment by A6GibKm** [comment]
By the way, for this usecase you dont need gskpaths, rectangles would be fine

**Comment by vixalien** [comment]
Hmm... I need to familiarize myself with the API
